From 39fd1d5c71a70d6f77dd7c3df43f9fd1308a104c Mon Sep 17 00:00:00 2001
Message-Id: <39fd1d5c71a70d6f77dd7c3df43f9fd1308a104c.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Fri, 14 Nov 2014 15:02:31 +0800
Subject: [PATCH 32/34] FOR_UPSTREAM [VPG]: drm/i915: restore HPD irq after
 runtime resume

During device runtime resume, we didn't restore the hotplug irq,
this causes HDMI hotplug not working.

Issue: GMINL-3811
Change-Id: I6307ee4e09c7de3bd12af8227f577a0ec8f75144
Signed-off-by: Hong Liu <hong.liu@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 9dd9b7e..33cdacf 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -5143,8 +5143,14 @@ void intel_runtime_pm_disable_interrupts(struct drm_device *dev)
 void intel_runtime_pm_restore_interrupts(struct drm_device *dev)
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
+	unsigned long flags;
 
 	dev_priv->pm.irqs_disabled = false;
 	dev->driver->irq_preinstall(dev);
 	dev->driver->irq_postinstall(dev);
+
+	spin_lock_irqsave(&dev_priv->irq_lock, flags);
+	if (dev_priv->display.hpd_irq_setup)
+		dev_priv->display.hpd_irq_setup(dev);
+	spin_unlock_irqrestore(&dev_priv->irq_lock, flags);
 }
-- 
1.7.9.5

